project_euler/problem_092/sol1.py

```diff
@@ -0,0 +1,53 @@
+"""
+Project Euler Problem 92: https://projecteuler.net/problem=92
+Name: Square digit chains
+A number chain is created by continuously adding the square of the digits in a number to
+form a new number until it has been seen before.
+For example,
+44 → 32 → 13 → 10 → 1 → 1
+85 → 89 → 145 → 42 → 20 → 4 → 16 → 37 → 58 → 89
+Therefore any chain that arrives at 1 or 89 will become stuck in an endless loop.
+What is most amazing is that EVERY starting number will eventually arrive at 1 or 89.
+How many starting numbers below ten million will arrive at 89?
+"""
+
+
+def solution(limit: int = 10000000) -> int:
+    """
+    Returns the number of starting numbers, up to a given limit, whose number chain
+    arrives at 89.
+    A number chain is created by continuously adding the square of the digits in a
+    number to form a new number until it has been seen before.
+    >>> solution(10)
+    7
+    >>> solution(350)
+    295
+    >>> solution(22222)
+    18803
+    >>> solution(10000000)
+    8581146
+    """
+
+    squared = {1: 1, 89: 89}
```

Comment:
If this is a cache then use `functools.lru_cache`

Reply:
Ah, my bad. This was a mistake on my side. Please remove the `lru_cache` as it is not needed here. But try to optimize this solution as it is taking 50 sec to run: https://github.com/TheAlgorithms/Python/pull/3509/checks?check_run_id=1360369919#step:5:25

Reply:
Sorry I have been very busy but I will try and optimise this weekend!